core/transport/src/main/proto/grpc/block_master.proto

```diff
@@ -168,10 +168,11 @@ message StorageList {
 message BlockHeartbeatPOptions {
   repeated grpc.Metric metrics = 1;
   map<string, int64> capacityBytesOnTiers = 2;
+  optional bool hasClusterId = 3;
```

Comment:
I dont think you need this field. You can check BlockHeartbeatPRequest.hasClusterId() directly because `clusterId` is an optional field. This method is automatically generated.

Reply:
done